### PR TITLE
mlflow model conversion from HF for image to image task

### PR DESCRIPTION
**Pull request by vadthyavath**
test pipelines: https://ml.azure.com/experiments/id/c41e6aae-f89c-4ccc-9d95-6c6b11572843/runs/tender_dress_w85qhxg5xn?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/rv-images-rc/providers/Microsoft.MachineLearningServices/workspaces/rv-images-ws&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

### CODE REVIEW

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:533]:
```suggestion
                    type=TextToImageMLFlowSchemaLiterals.INPUT_COLUMN_PROMPT_DATA_TYPE),
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:535]:
```suggestion
                    type=TextToImageMLFlowSchemaLiterals.INPUT_COLUMN_IMAGE_TYPE)
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:541]:
```suggestion
                    type=TextToImageMLFlowSchemaLiterals.OUTPUT_COLUMN_NSFW_FLAG_TYPE),
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:550]:
```suggestion
        """Initialize MLflow convertor for SD image to image models."""
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:565]: Don't need two doc strings

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/stable_diffusion_image_to_image_mlflow_wrapper.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/stable_diffusion_image_to_image_mlflow_wrapper.py:107]: nit: new line at the end of the file.

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:530]:
```suggestion
        """Return model signature for image-text to image models."""
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:521]:
```suggestion
    """MlfLow convertor base class for image-text to image models."""
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:526]:
```suggestion
        """Initialize MLflow convertor for image-text to image models."""
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/convertors.py:547]:
```suggestion
    """HF MlfLow convertor class for stable diffusion image-text to image models."""
```

File: assets/training/model_management/components/convert_model_to_mlflow/spec.yaml
Comment (rjaincc) [inline assets/training/model_management/components/convert_model_to_mlflow/spec.yaml:4]: why are we going from 15 to 17?

File: assets/models/system/stabilityai-stable-diffusion-xl-refiner-1-0/description.md
Comment (rjaincc) [inline assets/models/system/stabilityai-stable-diffusion-xl-refiner-1-0/description.md:3]: The link for HF model card is pointing to the base model, should be pointing to the refiner model

File: assets/models/system/stabilityai-stable-diffusion-xl-refiner-1-0/description.md
Comment (rjaincc) [inline assets/models/system/stabilityai-stable-diffusion-xl-refiner-1-0/description.md:9]: Are aka.ms links generated?

**Review by rjaincc — DISMISSED** [review]
Approving with minor comments. Please address these before merging.

File: assets/training/model_management/components/convert_model_to_mlflow/spec.yaml
Comment (vadthyavath) [reply assets/training/model_management/components/convert_model_to_mlflow/spec.yaml:4]: changed to 15!

**Review by MadhuM02 — APPROVED** [review]

**Review by MadhuM02 — DISMISSED** [review]

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/stable_diffusion_image_to_image_mlflow_wrapper.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/stable_diffusion_image_to_image_mlflow_wrapper.py:32]: Let's use ImageTexttoImage / image-text-to-image for consistency

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/config.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/config.py:21]: I think, would be better to call it "IMAGE_TEXT_TO_IMAGE"

File: assets/models/system/stabilityai-stable-diffusion-xl-refiner-1-0/spec.yaml
Comment (rjaincc) [inline assets/models/system/stabilityai-stable-diffusion-xl-refiner-1-0/spec.yaml:16]: We want to keep it "image-to-image" --> same as HuggingFace to align

**Review by rjaincc — CHANGES_REQUESTED** [review]
Please address the comments.

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/config.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/text_to_image/config.py:21]:
```suggestion
    IMAGE_TO_IMAGE = "image-to-image"
```

File: assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/config.py
Comment (rjaincc) [inline assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/config.py:43]:
```suggestion
    IMAGE_TO_IMAGE = "image-to-image"
```

**Review by MadhuM02 — APPROVED** [review]

**Review by rjaincc — APPROVED** [review]

**Pull request merged**
